apps/system/test/unit/update_manager_test.js

```diff
@@ -428,8 +428,6 @@ suite('system/UpdateManager', function() {
         assert.isFalse(UpdateManager.downloadViaDataConnectionDialog.
           classList.contains('visible'));
         assert.isFalse(MockCustomDialog.mShown);
-        // should not close all custom dialog when no custom dialog is opened
-        assert.isFalse(Service.request.calledWith('hideCustomDialog'));
```

Comment:
@gasolin I know it was a while ago (July) that you added this.

All the other show dialog functions were setting `UpdateManager._hasDialog` to `true` except `UpdateManager.showDownloadPrompt()`. In this PR I do the same in `UpdateManager.showDownloadPrompt()`. 

Now, when we handle the `home` and `holdhome` events, we know to only call `UpdateManager.cancelPrompt()` if there is a dialog showing.

Do you have any thoughts or recommendations?